src/Core/Parser/Entity/CollectionLogOperation/OperationsCollection.php

```diff
@@ -87,18 +87,18 @@ private function checkIsFoundEntitiesCacheOutdatedRecursive(
         foreach ($operationsCollection->operations as $operation) {
             if ($operation instanceof SingleEntitySearchOperation) {
                 $entity = $operation->call($rootEntityCollection);
-                if (is_null($operation->getEntityName()) && (is_null($entity) || $entity->entityCacheIsOutdated())) {
+                if (is_null($operation->getEntityName()) && (is_null($entity) || $entity->isEntityCacheOutdated())) {
                     continue;
                 }
                 if ($entity && !$entity::isEntityNameValid($entity?->getName())) {
                     continue;
                 }
 
                 $entityName = $entity?->getName();
-                $entityName = $entityName && $entity?->entityDataCanBeLoaded() ? $entityName : null;
+                $entityName = $entityName && $entity?->isEntityDataCanBeLoaded() ? $entityName : null;
                 if ($operation->getEntityName() !== $entityName) {
                     return true;
-                } elseif ($entity?->entityCacheIsOutdated() && $entity?->entityDataCanBeLoaded()) {
+                } elseif ($entity?->isEntityCacheOutdated() && $entity?->isEntityDataCanBeLoaded()) {
```

Comment:
This could be converted to just an if statement, which is clearer / lower complexity. 
```suggestion
                } 
                if ($entity?->isEntityCacheOutdated() && $entity?->isEntityDataCanBeLoaded()) {
```